prebuilt/NINJA_subgraph.py

```diff
@@ -74,4 +76,8 @@ def NinjaGraph(ru):
             '//cpp/frontend_flag_spec',
         ])
 
-    ru.souffle_cpp('mycpp/datalog/call-graph.dl')
+    # NOTE: To avoid failures on these in CI, these targets are only enabled if
+    # the user has the souffle executable in their PATH.
+    if subprocess.call('which souffle > /dev/null', shell=True) == 0:
+        ru.souffle_cpp('mycpp/datalog/call-graph.dl')
+        ru.souffle_cpp('deps/source.medo/souffle/smoke-test.dl')
```

Comment:
@andychu thoughts on this?

Reply:
If we are always prebuilding the datalog, can we simply have NO Ninja support at all for Souffle?

Logically it could be outside the build system

We could just have a shell script you run every time you change the datalog ...

---

I am not sure what the right tradeoff is yet -- we'll have to work with it

But if it is reasonable to work with, there's no rule that everything has to be in Ninja

the `prebuilt/` stuff is generally a special case -- it is already handled with shell scripts, outside of Ninja

Reply:
Oh right. I kept these ninja targets around because they were shorter than typing out `build/ninja-rules-datalog.sh compile_souffle ... ...`, but we could definitely achieve the same effect with a script... I'll push a commit to replace this with a script

Reply:
OK pushed this. Should be good to go now